src/common/mfu_io.c

```diff
@@ -516,7 +1012,31 @@ void mfu_getcwd(char* buf, size_t size)
     }
 }
 
-/* create directory, retry a few times on EINTR or EIO */
+int daos_mkdir(const char* dir, mode_t mode, mfu_file_t* mfu_file) {
+#ifdef DAOS_SUPPORT
+    char* name     = NULL;
+    char* dir_name = NULL;
+    parse_filename(dir, &name, &dir_name);
+    assert(dir_name);
+
+    /* Need to lookup parent directory in DFS */
+    dfs_obj_t *parent = NULL;
+    int rc = dfs_lookup(mfu_file->dfs, dir_name, O_RDWR, &parent, NULL, NULL);
+
+    /* only call mkdir if the dir_name is not the root DFS directory */
+    if (strcmp(dir_name, "/") != 0) {
+        rc = dfs_mkdir(mfu_file->dfs, parent, name, S_IRWXU, 0);
```

Comment:
Does this need a check to see if dfs_lookup succeeded?

Reply:
@adammoody well it is ok if the lookup doesn't succeed, since not all directories in DFS have a parent. If there is an error then the check in the function will catch it.

Reply:
Ok, let's add a comment about that, because I'll probably forget that detail later.